src/BaseIO.hpp

```diff
@@ -238,6 +238,12 @@ class BaseIO
   virtual Status createReferenceDataSet(
       const std::string& path, const std::vector<std::string>& references) = 0;
 
+  /**
+   * @brief Starts the recording process.
+   * @return The status of the operation.
+   */
+  virtual Status startRecording() = 0;
```

Comment:
We should also add a `stopRecording` method on `BaseIO` and `HDF5IO`. Otherwise we can start a recording in the IO but not stop it. This is also goes along with the comment about moving the `isRecording` flag to the IO.